sale_order_lot_selection/model/stock.py

```diff
@@ -11,7 +11,7 @@ class StockMove(models.Model):
     _inherit = 'stock.move'
 
     @api.model
-    def _prepare_procurement_from_move(self, move):
-        vals = super(StockMove, self)._prepare_procurement_from_move(move)
-        vals['lot_id'] = move.restrict_lot_id.id
+    def _prepare_procurement_from_move(self):
+        vals = super(StockMove, self)._prepare_procurement_from_move()
+        vals['lot_id'] = self.restrict_lot_id.id
```

Comment:
Yes, So I guess you should remove the api.model, to be coherent with native method